src/dtclient/kubernetes_settings.go

```diff
@@ -0,0 +1,254 @@
+package dtclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type postKubernetesSettings struct {
+	Label                           string `json:"label"`
+	ClusterIdEnabled                bool   `json:"clusterIdEnabled"`
+	ClusterId                       string `json:"clusterId"`
+	CloudApplicationPipelineEnabled bool   `json:"cloudApplicationPipelineEnabled"`
+	OpenMetricsPipelineEnabled      bool   `json:"openMetricsPipelineEnabled"`
+	Enabled                         bool   `json:"enabled"`
+	EventProcessingActive           bool   `json:"eventProcessingActive"`
+	EventProcessingV2Active         bool   `json:"eventProcessingV2Active"`
+	FilterEvents                    bool   `json:"filterEvents"`
+}
+
+type postKubernetesSettingsBody struct {
+	SchemaId      string                 `json:"schemaId"`
+	SchemaVersion string                 `json:"schemaVersion"`
+	Scope         string                 `json:"scope,omitempty"`
+	Value         postKubernetesSettings `json:"value"`
+}
+
+type monitoredEntitiesResponse struct {
+	TotalCount int               `json:"totalCount"`
+	PageSize   int               `json:"pageSize"`
+	Entities   []MonitoredEntity `json:"entities"`
+}
+
+type MonitoredEntity struct {
+	EntityId    string `json:"entityId"`
+	DisplayName string `json:"displayName"`
+	LastSeenTms int64  `json:"lastSeenTms"`
+}
+
+type GetSettingsResponse struct {
+	TotalCount int `json:"totalCount"`
+}
+
+type postSettingsResponse struct {
+	ObjectId string `json:"objectId"`
+}
+
+type getSettingsErrorResponse struct {
+	ErrorMessage getSettingsError `json:"error"`
+}
+
+type getSettingsError struct {
+	Code                 int
+	Message              string
+	ConstraintViolations constraintViolations
+}
+
+type constraintViolations []struct {
+	ParameterLocation string
+	Location          string
+	Message           string
+	Path              string
+}
+
+func (dtc *dynatraceClient) CreateKubernetesSetting(name, kubeSystemUUID, scope string) (string, error) {
+	if name == "" {
```

Comment:
checking the` name != ""` is not necessary, because its the name of the DynaKube, and it must have a name to exist in Kubernetes